src/Diagnostics.RuntimeHost/Controllers/Site/SitesController.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Diagnostics.ModelsAndUtils.Models;
+using Diagnostics.RuntimeHost.Models;
+using Diagnostics.RuntimeHost.Services;
+using Diagnostics.RuntimeHost.Services.SourceWatcher;
+using Diagnostics.RuntimeHost.Utilities;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace Diagnostics.RuntimeHost.Controllers
+{
+    [Produces("application/json")]
+    [Route(UriElements.SitesResource + UriElements.Diagnostics)]
+    public sealed class SitesController : SiteControllerBase
+    {
+        public SitesController(IStampService stampService, ISiteService siteService, ICompilerHostClient compilerHostClient, ISourceWatcherService sourceWatcherService, IInvokerCacheService invokerCache, IDataSourcesConfigurationService dataSourcesConfigService)
+            : base(stampService, siteService, compilerHostClient, sourceWatcherService, invokerCache, dataSourcesConfigService)
+        {
+        }
+
+        [HttpPost(UriElements.Query)]
+        public async Task<IActionResult> Post(string subscriptionId, string resourceGroupName, string siteName, [FromBody]CompilationBostBody<DiagnosticSiteData> jsonBody, string startTime = null, string endTime = null, string timeGrain = null)
+        {
+            if (jsonBody == null)
+            {
+                return BadRequest("Missing body");
+            }
+
+            if (string.IsNullOrWhiteSpace(jsonBody.Script))
+            {
+                return BadRequest("Missing script in body");
+            }
+
+            if(jsonBody.Resource == null)
+            {
+                return BadRequest("Missing Site data in body");
```

Comment:
This worries me, I think it is possible that we want to be able to call this without the site data in the body. For example, I was trying to test support center with these APIs, which would enable me to test generic detectors without access to ARM for that resource. However, if you enforce this, that would not be possible.

Reply:
the next change would be to fetch the data if its not present. I'll use Hawk's work when that gets checked in.